### PR TITLE
Modify rule S3443: LaYC format

### PR DESCRIPTION
**Pull request by sebastien-marichal**

### CODE REVIEW

File: rules/S3443/csharp/rule.adoc [hunk 1]
```diff
@@ -1,12 +1,28 @@
...
 
-If you call ``++GetType()++`` on a ``++Type++`` variable, the return value will always be ``++typeof(System.Type)++``. So there's no real point in making that call. The same applies to passing a type argument to ``++IsInstanceOfType++``. In both cases the results are entirely predictable.
+Calling `GetType` on a variable of type `Type` will always return the `Type` representation of `System.Type` (aka. `typeof(System.Type)`). This also applies to passing a `Type` argument to `IsInstanceOfType`.
 
+In both cases, the results are entirely predictable and should be avoided.
 
-=== Noncompliant code example
+=== Exceptions
+
+Calling `GetType` on `System.Type` is considered compliant to get an instance of `System.RuntimeType`, like in the following example:
```
Comment (cristian-ambrosini-sonarsource) [inline rules/S3443/csharp/rule.adoc:9]: Educational: so if I understood correctly:
```cs
string str = string.Empty;

Type strType = str.GetType(); // Compliant (System.String)
Type strTypeType = strType.GetType(); // Noncompliant (System.RuntimeType)
typeof(Type).GetType(); // Compliant (System.RuntimeType)
```

So the exception of this rule is specifically the line `typeof(Type).GetType();` to allow one way to retrieve System.RuntimeType without raising the issue. Am I right?

File: rules/S3443/csharp/rule.adoc [hunk 2]
```diff
@@ -1,12 +1,28 @@
...
+Make sure the usage of `GetType` or `IsInstanceOfType` is invoked with correct variable or argument type.
+
+=== Code examples
+
+==== Noncompliant code example
+
+[source,csharp,diff-id=1,diff-type=noncompliant]
+----
 var intType = typeof(int);
 var runtimeType = intType.GetType(); // Noncompliant, always typeof(System.RuntimeType)
```
Comment (cristian-ambrosini-sonarsource) [inline rules/S3443/csharp/rule.adoc:27]: This first example is not fixed in the compliant example, right?
WDYT about a test like the following?

```cs
void ExamineSystemType(string str, Type type) 
{
    type.GetType(); // Noncompliant
    
    if (type.IsInstanceOfType(typeof(string))) // Noncompliant
    { /* ... */ }
}
```
and then the compliant version:
```cs
void ExamineSystemType(string str, Type type) 
{
    str.GetType(); // Noncompliant
    
    if (str.IsInstanceOfType(typeof(string))) // Noncompliant
    { /* ... */ }
}
```

**Review by cristian-ambrosini-sonarsource — CHANGES_REQUESTED** [review]
Looks good! I suggested a few changes

File: rules/S3443/csharp/rule.adoc [hunk 3]
```diff
@@ -1,12 +1,28 @@
 == Why is this an issue?
 
-If you call ``++GetType()++`` on a ``++Type++`` variable, the return value will always be ``++typeof(System.Type)++``. So there's no real point in making that call. The same applies to passing a type argument to ``++IsInstanceOfType++``. In both cases the results are entirely predictable.
+Calling `GetType` on a variable of type `Type` will always return the `Type` representation of `System.Type` (aka. `typeof(System.Type)`). This also applies to passing a `Type` argument to `IsInstanceOfType`.
```
[the remark this answers is not in the document]
Comment (sebastien-marichal) [reply rules/S3443/csharp/rule.adoc:3]: I wanted to have a clearer sentence, I think the old version is not easy to understand for junior developers. And I agree, `type` is used too many times.

**Review by cristian-ambrosini-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**